File: tests/fixtures/generic/clean/taint_codeinj_clean.cs
// @audit-fixture
// @rule: taint_codeinj
// @category: security
// @expected: clean
public class ScriptController : Controller {
    private static readonly string[] ALLOWED = { "report", "status" };
    public IActionResult Run() {
        var op = Request.Query["op"].ToString();
        if (!ALLOWED.Contains(op)) return BadRequest();
        return Ok();
    }
}

File: tests/fixtures/generic/clean/taint_codeinj_clean.java
// @audit-fixture
// @rule: taint_codeinj
// @category: security
// @expected: clean
public class EvalServlet extends HttpServlet {
    private static final Set<String> ALLOWED = Set.of("report", "status");
    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws Exception {
        String op = request.getParameter("op");
        if (!ALLOWED.contains(op)) { response.sendError(400); return; }
        // Exécuter l'opération depuis une liste blanche
    }
}
